code-challenges/401challenges/src/main/java/challenges/stacksandqueues/Queue.java

```diff
@@ -0,0 +1,79 @@
+package challenges.stacksandqueues;
+
+public class Queue {
+    private Node front;
+    private Node back;
+    private int size;
+
+    public Node getFront() {
+        return front;
+    }
+
+    public void setFront(Node front) {
+        this.front = front;
+    }
+
+    public int getSize() {
+        return size;
+    }
+
+    public void setSize(int size) {
+        this.size = size;
+    }
+
+    public Node getBack() {
+        return back;
+    }
+
+    public void setBack(Node back) {
+        this.back = back;
+    }
+
+    public Queue(){
+        this.front = null;
+        this.back = null;
+        this.size = 0;
+    }
+
+    public void enqueue(int value){
+        try
+        {
+            Node node = new Node(value);
+            node.next = null;
+            if(this.front == null){
+                this.front = node;
+                this.back = node;
+                this.size++;
+            } else {
+                Node last = this.back;
+                last.next = node;
+                this.back = node;
+                this.size++;
+            }
+
+        } catch(IllegalArgumentException ex){
+            System.out.println("You either didn't pass anything in or you didn't pass in an Integer");
+        }
+    }
+
+    public int dequeue(){
+        if(this.front == null){
+            return 0;
```

Comment:
How does the user know that 0 means that the Queue is empty? What is 0 is the value of the Node being dequeued?